public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+  by following this plan:
+
+  * Reset the type of the `heroForm` property to `FormGroup`. 
+  * Inject a `FormBuilder` into the constructor.
+  * Add a new `createForm` method that uses the `FormBuilder` to define the `heroForm`.
+  * Call `createForm` in the constructor.
+
+  The revised `HeroDetailComponent` looks like this:
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'v2','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  `FormBuilder.group` is a factory method that creates a `FormGroup`. &nbsp;
+  `FormBuilder.group` takes an object whose keys and values are `FormControl` names and their definitions. 
+  In this example, the `name` control is defined by its initial data value, an empty string.
+
+  Defining a group of controls in a single object makes for a compact, readable style.
+  It beats writing an equivalent series of `new FormControl(...)` statements.
+
+  The form looks great. But does it work? 
+  When the user enters a name, where does the value go? 
+  
+  The value goes into the **_form model_** that backs the group's `FormControls`.
+  To see the form model, add the following line after the 
+  closing `form` tag in the `hero-detail.component.html`:
+
++makeExample('reactive-forms/ts/app/hero-detail-2.component.html', 'form-value-json','app/hero-detail.component.html')(format=".")
+
+:marked
+  The `heroForm.value` returns the _form model_.
+  Piping it through the `JsonPipe` renders the model as JSON in the browser:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/json-output.png" width="400px" alt="JSON output")
+
+:marked
+  The initial `name` property value is the empty string. 
+  Type into the _name_ input box and watch the keystokes appear in the JSON.
+
+  ### More FormControls
+  A hero has more than a name. A hero has an address too. 
+  Add some address `FormControls` to the `heroForm` as follows.
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'v3','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  Then add corresponding markup in `hero-detail.component.html` 
+  within the `form` element.
+
++makeExample('reactive-forms/ts/app/hero-detail-3.component.html', '','app/hero-detail.component.html')(format=".")
+
+.alert.is-helpful
+  :marked
+    *Reminder*: Ignore the many mentions of `form-group` and `form-control`in this markup.
+    Those are _bootstrap_ CSS classes that Angular itself ignores.
+    Pay attention to the `formGroupName` and `formControlName` attributes.
+    They are the Angular directives that bind the HTML controls to the 
+    Angular `FormGroup` and `FormControl` properties in the component class.
+
+.l-main-section
+a#grouping
+:marked
+  ### Group FormControls
+
+  This form is getting big and unwieldy.
+  You can make it more maintainable by grouping some of the related `FormControls` into another `FormGroup`.
```

Comment:
I have never thought of grouping as a technique for making a form more maintainable. In some respects, it makes it a bit more complex. To me, the real reasons to group controls are:
1) To better match the hierarchical structure of the data model.
2) To track validation and state for a logically related set of controls.